fuse_core/src/async_motion_model.cpp

```diff
@@ -101,10 +101,17 @@ void AsyncMotionModel::start()
 
 void AsyncMotionModel::stop()
 {
-  auto callback = boost::make_shared<CallbackWrapper<void>>(std::bind(&AsyncMotionModel::onStop, this));
-  auto result = callback->getFuture();
-  callback_queue_.addCallback(callback, reinterpret_cast<uint64_t>(this));
-  result.wait();
+  if (ros::ok())
+  {
+    auto callback = boost::make_shared<CallbackWrapper<void>>(std::bind(&AsyncMotionModel::onStop, this));
+    auto result = callback->getFuture();
+    callback_queue_.addCallback(callback, reinterpret_cast<uint64_t>(this));
+    result.wait();
+  }
+  else
+  {
+    onStop();
```

Comment:
This has the strange side effect of executing the `onStop()` method from a different thread than normal. The async spinner _should_ stop executing callbacks when ros::ok() goes false...which I suppose is the problem...so using a different thread _should_ be fine.

Would it make any sense to be explicit about it and add a `spinner_.stop()` call first? Maybe that's overkill?

Reply:
If I call `spinner_.stop()` that would guarantee the last callback that was running finishes. So we probably have to do so, so `onStop()` is not running in parallel with any user defined callback in the child class. :thinking: 

I'm going to make this change, since it really looks like it's important. Let me know your thoughts anyway.

Reply:
Done.